simulators/history/rpc-compat/src/main.rs

```diff
@@ -49,125 +49,192 @@ async fn run_suite(host: Simulation, suite: Suite) {
+                    name: "portal_historyDeleteEnr None Found".to_string(),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_history_delete_enr_non_present,
+                    environments: None,
+                    test_data: None,
+                    clients: vec![client.clone()],
+                }
+            ).await;
+
+            test.run(
+                NClientTestSpec {
+                    name: "portal_historyDeleteEnr ENR Found".to_string(),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_history_delete_enr_enr_present,
+                    environments: None,
+                    test_data: None,
+                    clients: vec![client.clone()],
+                }
+            ).await;
+
+            test.run(
+                NClientTestSpec {
+                    name: "portal_historyLookupEnr None Found".to_string(),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_history_lookup_enr_non_present,
+                    environments: None,
+                    test_data: None,
+                    clients: vec![client.clone()],
+                }
+            ).await;
+
+            test.run(
+                NClientTestSpec {
+                    name: "portal_historyLookupEnr ENR Found".to_string(),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_history_lookup_enr_enr_present,
+                    environments: None,
+                    test_data: None,
+                    clients: vec![client.clone()],
+                }
+            ).await;
+
+            test.run(
+                NClientTestSpec {
+                    name: "portal_historyLookupEnr Local Enr".to_string(),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_history_lookup_enr_local_enr,
+                    environments: None,
+                    test_data: None,
+                    clients: vec![client.clone()],
+                }
+            ).await;
+
+            test.run(
+                NClientTestSpec {
+                    name: "portal_historyRecursiveFindContent Content Absent".to_string(),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_recursive_find_content_content_absent,
+                    environments: None,
+                    test_data: None,
+                    clients: vec![client.clone()],
+                }
+            ).await;
+        }
     }
 }
 
 dyn_async! {
-    async fn test_node_info<'a> (client: Client) {
-       let response = client
-            .rpc
-            .node_info().await;
+    async fn test_node_info<'a>(clients: Vec<Client>, _: Option<Vec<(String, String)>>) {
```

Comment:
Why we are changing here the function signature to accept `Vec<Client>` if we are only going to use the first client?

How hard is it to add environments to `ClientTestSpec`?

Reply:
I want to remove ``ClientTestSpec`` and ``TwoClientTestSpec``. I don't think anything is "hard" per say, but I don't want to maintain multiple function signatures which all do the exact same thing. If there is ever a new feature. Adding new features to 3 almost duplicate function signatures, when it already takes 1-3 weeks for me to get a review for my Portal-Hive changes I don't want bigger PR becauses I am maintaining duplicate logic. Big PR's are scary.

Another reason for this is when we move to ``ethereum/hive``. ``hivesim-rs`` and simulators that use it will have to be updated in seperate PR's for every change. So having less code in ``hivesim-rs`` that has to be reviewed or that can delay the next PR is ideal. As then there is less chances bugs or issues can go through.

``NClientTestSpec`` can accomplish everything instead of requiring multiple signatures to support our test-suites we only need 1.

> How hard is it to add environments to ClientTestSpec instead of using NClientTestSpec?

Also ``environments`` isn't the only thing that would need to be ported to ``ClientTestSpec``. TestData would have to be done too. If I ever change the format for test data I also don't want to copy-paste the new logic to support ``ClientTestSpec`` and ``TwoClientTestSpec`` which is exactly the reason I want to depreicate them. ``NClientTestSpec`` can just replace them and remove the need to support 2 different function signatures.

Reply:
Cool, fair enough :+1: .